configure

```diff
@@ -367,23 +372,19 @@ detect_cpp() {
           -I "$FLAG_readline/include" \
           -l readline; then
         echo '#define HAVE_READLINE 1'
+        return 0
```

Comment:
Thanks for fixing this!

Looks like there are a lot of problems in this script -- I took a look at the git history 

-----

Are you up for a bigger refactoring?  You mentioned the problem with the duplication between with `detect_readline` and `detect_cpp`, and I agree

I think the problem is that there should be a SINGLE detect_readine that uses the EARLY return style.  Right now the styles are mixed, whic is probably why the code is buggy

Also the exit code doesn't seem to mean anything, so it should just `return`

I think it should look like

```
if test $FLAG_without_readline ...
  echo ...
  return

if test $FLAG_readline ...
  if cc_quiet ...
     echo ...
  return 

if cc_quiet ...
  echo ...
  return

if test $FLAG_with_readline ...
  die ...

echo 'readline not detected'

```


It's kind of annoying to have all these flags, but AFAIK they were based on what GNU autoconf has ??  Open to suggestions


I'm OK with a bigger change because we're focused on  making the C++ tarball work nicely now

I can merge this as is since it looks like a good fix, but I think we should eventually have the refactoring

Reply:
Or actually there is a slight bug in that logic, because it can return before printing the "not detected" case!  

I guess `detect_readline` could be defined to print success or die, and then another function could print the "not detected" case

There is a unit test in `./configure-test.sh` that could possibly help with this

Reply:
I did some refactoring to not check for readline twice, but it required setting a flag inside `detect_cpp` that modifies how `detect_and_echo_vars` acts. The code is written in a way that it echos values to be written to a file and needs to handle multiple cases: `without-readline` passed, `with-readline` passed and no `readline` flag, `with-readline` passed with a `readline` flag, and `readline` passed without `with-readline`.